torch/distributed/fsdp/_flat_param.py

```diff
@@ -1480,9 +1510,13 @@ def unshard_grad(self):
             self._check_sharded(flat_param.grad)
             flat_param._saved_grad_shard = flat_param.grad  # type: ignore[attr-defined]
             sharded_grad = flat_param._saved_grad_shard  # type: ignore[attr-defined]
-        dist.all_gather_into_tensor(
-            padded_unsharded_grad, sharded_grad, self.process_group
+
+        pg = (
+            self._fake_process_group
+            if self._use_fake_all_gather
+            else self.process_group
         )
```

Comment:
This `unshard_grad` should only be called from `summon_full_params(with_grads=True)`, which is not normally called during training. I am not sure we have to fakify this one.

Reply:
Don't fakify this one too.